internal-export-file/export-file-stix/src/export-file-stix.py

```diff
@@ -49,51 +50,89 @@ def _process_message(self, data):
                 + ") to "
                 + file_name
             )
-        else:
-            list_params = data["list_params"]
-            self.helper.log_info(
-                "Exporting list: "
-                + entity_type
-                + "/"
-                + export_type
-                + " to "
-                + file_name
-            )
-            bundle = self.helper.api_impersonate.stix2.export_list(
-                entity_type,
-                list_params["search"],
-                list_params["filters"],
-                list_params["orderBy"],
-                list_params["orderMode"],
-                max_marking,
-                list_params.get("types"),
-                list_params.get("elementId"),
-                list_params.get("fromId"),
-                list_params.get("toId"),
-                list_params.get("elementWithTargetTypes"),
-                list_params.get("fromTypes"),
-                list_params.get("toTypes"),
-                list_params.get("relationship_type"),
-            )
+
+        else:  # export_scope = 'query' or 'selection'
+            if "element_id" in data:
+                element_id = data["element_id"]
+            else:
+                element_id = None
+
+            if export_scope == "selection":
+                selected_ids = data["selected_ids"]
+                list_filters = "selected_ids"
+                entities_list = []
+
+                for entity_id in selected_ids:
+                    entity_data = self.helper.api_impersonate.stix_domain_object.read(
+                        id=entity_id
+                    )
+                    if entity_data is None:
```

Comment:
It's not possible to retrieve the type of the entity for this for each ?
I mean, can we export a list with SDO and SCO or SDO and relationship ?

If not, maybe we can give direclty the parent type of the export entities (entity_type property).

Reply:
I used how the code was already done. Maybe we can do this differently.